resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -274,22 +292,28 @@ private[spark] object Client {
     val appArgs = args.drop(2)
     run(sparkConf, mainAppResource, mainClass, appArgs)
   }
-
   def run(
       sparkConf: SparkConf,
       mainAppResource: String,
       mainClass: String,
       appArgs: Array[String]): Unit = {
-    val sparkJars = sparkConf.getOption("spark.jars")
+    val isPython = mainAppResource.endsWith(".py")
+    val sparkJars = if (isPython) Array.empty[String] else {
+      sparkConf.getOption("spark.jars")
       .map(_.split(","))
       .getOrElse(Array.empty[String]) ++
       Option(mainAppResource)
         .filterNot(_ == SparkLauncher.NO_RESOURCE)
-        .toSeq
+        .toSeq }
     val launchTime = System.currentTimeMillis
     val sparkFiles = sparkConf.getOption("spark.files")
       .map(_.split(","))
       .getOrElse(Array.empty[String])
+    val pySparkFiles: Array[String] = if (isPython) {
```

Comment:
I'm starting to wonder if we want our arguments to be similar to CLI arguments. We can then adjust `SparkSubmit.scala` to conform to the contract here.

For example, we could expect our main method to have arguments like this:

`org.apache.spark.deploy.kubernetes.Client --primary-resource <resource> --py-files <pyfiles>`.